src/Keystore/AESParameters.cpp

```diff
@@ -14,18 +14,30 @@ using namespace TW;
 
 namespace TW::Keystore {
 
-AESParameters::AESParameters() {
-    iv = Data(blockSize, 0);
-    random_buffer(iv.data(), blockSize);
+AESParameters::AESParameters(AESSize keyLength) : mKeyLength(keyLength) {
+    // https://www.reddit.com/r/crypto/comments/30x5xg/what_length_should_the_iv_be_for_aes256ctr/
+    // First off, AES 128 uses a 128 bit key.
+    // So if you're using AES 256, you're using a 256 bit key.
+    // Let's not confuse the block length with key length here.
+    // For AES, your block length is always going to be 128 bits/16 bytes regardless of the key length used
+    iv = Data(A128, 16);
+    random_buffer(iv.data(), 16);
```

Comment:
Why byte 16 is used as initial value?
Also, in one place `A128` is used as size, in the other `16`.
This would make more sense to me:
```
    iv = Data(A128, 0);
    random_buffer(iv.data(), iv.size());
```

Reply:
> Why byte 16 is used as initial value? Also, in one place `A128` is used as size, in the other `16`. This would make more sense to me:
> 
> ```
>     iv = Data(A128, 0);
>     random_buffer(iv.data(), iv.size());
> ```

IV is always 16 bytes long: https://www.reddit.com/r/crypto/comments/30x5xg/what_length_should_the_iv_be_for_aes256ctr/

Reply:
But the `16`, the second parameter is not a length, but default value.
Doesn't really matter as the next line overwrites it, but it is confusing, and not intended I guess.
Let's just use 0 as default value.